documentation/directory-structure.md

```diff
@@ -0,0 +1,245 @@
+| Short Link: | [aka.ms/azsdk/spec-dirs](https://aka.ms/azsdk/spec-dirs) |
+|--|--|
+
+<!-- 
+Table of contents generated with VSCode Markdown All in One:
+https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one 
+-->
+
+- [`specification` directory structure](#specification-directory-structure)
+  - [`specification` folder](#specification-folder)
+  - [`resource-manager` and `data-plane` folders](#resource-manager-and-data-plane-folders)
+  - [AutoRest configuration `README.md` files](#autorest-configuration-readmemd-files)
+  - [`stable` and `preview` folders](#stable-and-preview-folders)
+  - [A complete example directory structure of `specification/AzureTeamFolder`](#a-complete-example-directory-structure-of-specificationazureteamfolder)
+  - [Naming guidelines for `specification` folder contents](#naming-guidelines-for-specification-folder-contents)
+  - [`specification/common-types`](#specificationcommon-types)
+  - [Advanced scenario: service group](#advanced-scenario-service-group)
+    - [Service group `common-types`](#service-group-common-types)
+    - [Versioning services in a service group](#versioning-services-in-a-service-group)
+  - [Legacy layouts and OpenAPI specs](#legacy-layouts-and-openapi-specs)
+  - 
+# `specification` directory structure
+
+
+This article describes the directory structure / folder layout of the `specification` folder.
+You may be also interested in following:
+- [Specification index]
+- [Resource Provider list]
+
+> [!IMPORTANT]
+> The layout described in this article is strictly enforced. There exist some exceptions for historical reasons. These exceptions are not allowed going forward.
+
+## `specification` folder
+
+The `specification` folder is the root folder for all service specifications.
+
+Each child of the `specification` folder corresponds to a `service` specification for given Azure team. Here we denote such folder as `AzureTeamFolder`.
+
+Given `AzureTeamFolder` has following structure:
```

Comment:
In other places we refer to these as "service families". There are plenty of cases were they are not one azure team.

Reply:
I lean towards keeping it to Azure Team. It is a bit vague on purpose, I think. We have bunch of other terms, like: service, service family, service group, spec family. I don't want to muddle the waters here more. Plus Azure team was suggested by @JeffreyRichter. 

In this document I decided to continue using 'service group' in this article. I think we should change the TypeSpec dir structure doc to also use 'service group' instead of 'service family'.

Are there any other places that you know of that use 'service family' or 'spec family' ?

Reply:
We call it families in https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.mdn and I also call it just service in https://azure.github.io/azure-sdk/policies_repostructure.html. I'm Ok with calling it team but I feel like team is more of an org construct which is something we try to avoid.

Reply:
@weshaggard I think in practice the specification dir subfolders are in fact loosely corresponding to our org structure.

As for service families - I think we should rewrite the TS doc to say service group instead. As part of:
- https://github.com/Azure/azure-sdk-tools/issues/8028